linnefromice/webfront-practice-2
# topic/form

- add screen
- add libs - @mui/lab, dayjs, react-dropzone
- create dropzone component
- fix
- update matter form
- adjust styles - label's font size & dropzone size
- add lib - react-hook-form
- fix style - dropzone
- refactor - extract textfield, radiogroup, dropzone
- refactor - move components to other files
- apply react-hook-form to components (textfield, radiogroup)
- apply react-hook-form to dropzone made by react-dropzone
- use textfield to input date instead of datepicker
- apply react-hook-form to textfield to input amount
- apply react-hook-form to textfield/amountfield
- fix - Each child in a list should have a unique key prop
- fix
- update textfield component - separate design / form
- add storybook components for uiParts/organisms
- apply validation (react-hook-form) to textfield
- add debug code
- fix textfield - move caption to component to have design from formed component
- apply validation (react-hook-form) to radiogroup
- modify dropzone - add wrapped component
- fix screen
- update design - set required icon
- fix textfield - label when required (not use required props)
- update screen - disable submit button except first time
- add storybook/tests for screens/paidMemberMng/matterForm/create
- add tests for screens/paidMemberMng/matterForm/create - case for submitting / vaidations
- update tests
- update tests
- refactor - separate header from CreateForm (add FirstForm)
- modify test
- add navigation
- save form state
- add lib - @testing-library/react-hooks
- add test for hooks to navigation
- fix
- add SelectContractTypeForm
- update test for SelectContractTypeForm - add case to confirm submit function
- update navigation function
- fix
- add ContractTypeForm
- add PaymentMethodForm
- add OnboardingForm
- add function - judge contract type page to display
- select form in ContractTypeForm
- add backPage function for forms
- update tests - backPage function for forms
- refactor - organize types/props
- add back page button to forms & update tests/storybook component
- add base of ContractTypeForm (for ChiraCeo)
- update ContractTypeForm - add radio inputs
- fix
- update storybook component for ContractTypeForm
- update tests for ContractTypeForm forms - check title & items displayed
- update tests for ContractTypeForm forms - check to submit failed
- update tests for ContractTypeForm - add case to input required items & submit
- update ContractTypeForm - select form in accordance with payment method
- update tests fir ContractTypeForm - add storybook components
- update ContractTypeForm - implement form for credit card
- refactor
- update tests for ContractTypeForm - add case to check items & submit in CreditCardForm
- update tests for ContractTypeForm - add case to check items in CreditCardForm
- update OnboardingForm - add items & update tests
- add confimation view (temp)
- fix
- fix
- add todos for not implemented tests

todos

- save form & reflect current form
- confirmation page
- list view page
- jsdoc
